Build error: MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.Runtime.CoreCLR' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]

I'm trying to build Dotnet from the release/2.2 branch on a Gentoo box, but run into issues. I'm not (yet) familiar with your project structure, please tell me if I'm doing something wrong or reporting this in the wrong place. For what it's worth, I observe nearly identical errors with the 2.1 branch.

To avoid errors caused by stale Nuget files, HOME is set to an empty directory. The top-level directory is a clean checkout of the current `release/2.2` branch from the `source-build` repository.

When I run `./build.sh` from the top-level directory of the repository eventually leads to this error output:

```
  Generating symbol file for System.Private.CoreLib.dll
  /var/tmp/dotnet/source-build/src/coreclr/bin/Product/Linux.x64.Release/crossgen /Platform_Assemblies_Paths /var/tmp/dotnet/source-build/src/coreclr/bin/Product/Linux.x64.Release /CreatePerfMap /var/tmp/dotnet/source-build/src/coreclr/bin/Product/Linux.x64.Release /var/tmp/dotnet/source-build/src/coreclr/bin/Product/Linux.x64.Release/System.Private.CoreLib.dll
  Microsoft (R) CoreCLR Native Image Generator - Version 4.5.30319.0
  Copyright (c) Microsoft Corporation.  All rights reserved.
  
  Successfully generated perfmap for native assembly '/var/tmp/dotnet/source-build/src/coreclr/bin/Product/Linux.x64.Release/System.Private.CoreLib.dll'.
  Generating nuget packages for Linux
  DistroRid is gentoo.-x64
  ROOTFS_DIR is 
  Running init-tools.sh
  Tools are already initialized
  Running: /var/tmp/dotnet/source-build/src/coreclr/Tools/dotnetcli/dotnet /var/tmp/dotnet/source-build/src/coreclr/Tools/run.exe /var/tmp/dotnet/source-build/src/coreclr/config.json build -Project=/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/packages.builds -MsBuildEventLogging=/l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log -MsBuildLog=/flp:Verbosity=normal;LogFile=/var/tmp/dotnet/source-build/src/coreclr/bin/Logs/Nuget_Release.log -BuildTarget -__IntermediatesDir=/var/tmp/dotnet/source-build/src/coreclr/bin/obj/Linux.x64.Release -__RootBinDir=/var/tmp/dotnet/source-build/src/coreclr/bin -BuildNugetPackage=false -UseSharedCompilation=false -__DoCrossArchBuild=0 -BuildArch=x64 -BuildType=Release -BuildOS=Linux -PortableBuild=false -- /p:DotNetPackageVersionPropsPath=/var/tmp/dotnet/source-build/bin/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/var/tmp/dotnet/source-build/bin/obj/x64/Release/RestoreSources.props /p:DotNetOutputBlobFeedDir=/var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/
  Running: /var/tmp/dotnet/source-build/src/coreclr/Tools/msbuild.sh /nologo /verbosity:minimal /clp:Summary  /l:BinClashLogger,Tools/Microsoft.DotNet.Build.Tasks.dll;LogFile=binclash.log  /var/tmp/dotnet/source-build/src/coreclr/src/.nuget/packages.builds /p:__BuildType=Release /p:__BuildArch=x64 /p:__BuildOS=Linux /p:__RootBinDir=/var/tmp/dotnet/source-build/src/coreclr/bin /p:__IntermediatesDir=/var/tmp/dotnet/source-build/src/coreclr/bin/obj/Linux.x64.Release /p:__DoCrossArchBuild=0  /maxcpucount /p:RestoreDefaultOptimizationDataPackage=false /p:UsePartialNGENOptimization=false /p:PortableBuild=false  /flp:Verbosity=normal;LogFile=/var/tmp/dotnet/source-build/src/coreclr/bin/Logs/Nuget_Release.log     /p:BuildNugetPackage=false /t:Build /p:UseSharedCompilation=false /p:DotNetPackageVersionPropsPath=/var/tmp/dotnet/source-build/bin/obj/x64/Release/PackageVersions.props /p:DotNetRestoreSourcePropsPath=/var/tmp/dotnet/source-build/bin/obj/x64/Release/RestoreSources.props /p:DotNetOutputBlobFeedDir=/var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.Runtime.CoreCLR' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.TestHost' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.Native' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.Jit' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.ILAsm' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
    Microsoft.NETCore.Jit -> /var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/packages/specs/Microsoft.NETCore.Jit.nuspec
    Microsoft.NETCore.ILAsm -> /var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/packages/specs/Microsoft.NETCore.ILAsm.nuspec
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.ILDAsm' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
    Microsoft.NETCore.Native -> /var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/packages/specs/Microsoft.NETCore.Native.nuspec
    Microsoft.NETCore.TestHost -> /var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/packages/specs/Microsoft.NETCore.TestHost.nuspec
    Microsoft.NETCore.Runtime.CoreCLR -> /var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/packages/specs/Microsoft.NETCore.Runtime.CoreCLR.nuspec
    Microsoft.NETCore.ILDAsm -> /var/tmp/dotnet/source-build/bin/obj/x64/Release/blob-feed/packages/specs/Microsoft.NETCore.ILDAsm.nuspec
  
  Build FAILED.
  
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.Runtime.CoreCLR' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.TestHost' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.TestHost/Microsoft.NETCore.TestHost.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.Native' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.Jit' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.Jit/Microsoft.NETCore.Jit.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.ILAsm' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILAsm/Microsoft.NETCore.ILAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: The "NugetPack" task failed unexpectedly. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018: System.Exception: The package ID 'runtime.gentoo.-x64.Microsoft.NETCore.ILDAsm' contains invalid characters. Examples of valid package IDs include 'MyPackage' and 'MyPackage.Sample'. [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.Validate(Manifest manifest) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at NuGet.Packaging.Manifest.ReadFrom(Stream stream, Func`2 propertyProvider, Boolean validateSchema) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.GetManifest(String nuspecPath, Func`2 nuspecPropertyProvider, Boolean isPackedPackage) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.NuGetPack.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
  /var/tmp/dotnet/source-build/src/coreclr/Tools/Packaging.targets(1242,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/var/tmp/dotnet/source-build/src/coreclr/src/.nuget/Microsoft.NETCore.ILDAsm/Microsoft.NETCore.ILDAsm.pkgproj]
      0 Warning(s)
      6 Error(s)
  
  Time Elapsed 00:00:10.56
  Command execution failed with exit code 1.
  ERROR: An error occured in /var/tmp/dotnet/source-build/src/coreclr/Tools/dotnetcli/dotnet /var/tmp/dotnet/source-build/src/coreclr/Tools/run 18. Check 18 logs under /var/tmp/dotnet/source-build/src/coreclr.
  Failed to generate Nuget packages.
  
  'coreclr' failed during build.
  See '/var/tmp/dotnet/source-build/bin/logs/coreclr.log' for more information.
  'known-good' failed during build.
```

I don't understand the "Check 18 logs" bit, but here's the full contents of `/var/tmp/dotnet/source-build/bin/logs/coreclr.log`:
 <https://gist.github.com/eroen/d48f0c139e00198a6cb0ef728d4ac3cb>

--- a/init-distro-rid.sh
+++ b/init-distro-rid.sh
@@ -56,7 +56,13 @@ initNonPortableDistroRid()
                     VERSION_ID=${VERSION_ID%.*}
                 fi
 
-                nonPortableBuildID="${ID}.${VERSION_ID}-${buildArch}"
+                if [ -z ${VERSION_ID+x} ]; then
+                        # Rolling release distros do not set VERSION_ID, so omit
+                        # it here to be consistent with everything else.
+                        nonPortableBuildID="${ID}-${buildArch}"
+                else
+                        nonPortableBuildID="${ID}.${VERSION_ID}-${buildArch}"
+                fi
             fi
             
         elif [ -e "${rootfsDir}/etc/redhat-release" ]; then

